trustfall_core/src/ir/mod.rs

```diff
@@ -94,6 +104,9 @@ pub struct IREdge {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub parameters: Option<Arc<EdgeParameters>>,
 
+    /// Indicating if this edge is optional.
+    ///
+    /// This would correspond to `@optional` in GraphQL.
```

Comment:
```suggestion
    /// Corresponds to the `@optional` directive.
```
I mentioned this in #142 -- let's try to uphold the distinction between Trustfall and GraphQL as much as possible, and further distinguish over time.